plugins/mapper-annotated-text/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/AnnotatedTextHighlighterTests.java

```diff
@@ -59,78 +60,97 @@
+            FieldType ft = new FieldType(TextField.TYPE_STORED);
+            if (randomBoolean()) {
+                ft.setIndexOptions(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS);
+            } else {
+                ft.setIndexOptions(IndexOptions.DOCS_AND_FREQS);
+            }
+            ft.freeze();
+            Document doc = new Document();
+            for (String input : markedUpInputs) {
+                Field field = new Field(fieldName, "", ft);
+                field.setStringValue(input);
+                doc.add(field);
+            }
+            iw.addDocument(doc);
+            reader = iw.getReader();
+            IndexSearcher searcher = newSearcher(reader);
+            iw.close();
+
+            AnnotatedText[] annotations = new AnnotatedText[markedUpInputs.length];
+            for (int i = 0; i < markedUpInputs.length; i++) {
+                annotations[i] = AnnotatedText.parse(markedUpInputs[i]);
+            }
+            AnnotatedHighlighterAnalyzer hiliteAnalyzer = new AnnotatedHighlighterAnalyzer(wrapperAnalyzer);
+            hiliteAnalyzer.setAnnotations(annotations);
+            AnnotatedPassageFormatter passageFormatter = new AnnotatedPassageFormatter(new DefaultEncoder());
+            passageFormatter.setAnnotations(annotations);
+
+            ArrayList<Object> plainTextForHighlighter = new ArrayList<>(annotations.length);
+            for (int i = 0; i < annotations.length; i++) {
+                plainTextForHighlighter.add(annotations[i].textMinusMarkup);
+            }
+
+            TopDocs topDocs = searcher.search(new MatchAllDocsQuery(), 1, Sort.INDEXORDER);
+            assertThat(topDocs.totalHits.value, equalTo(1L));
+            String rawValue = Strings.collectionToDelimitedString(plainTextForHighlighter, String.valueOf(MULTIVAL_SEP_CHAR));
+            CustomUnifiedHighlighter highlighter = new CustomUnifiedHighlighter(
+                    searcher,
+                    hiliteAnalyzer,
+                    UnifiedHighlighter.OffsetSource.ANALYSIS,
+                    passageFormatter,
+                    locale,
+                    breakIterator,
+                    "index",
+                    "text",
+                    query,
+                    noMatchSize,
+                    expectedPassages.length,
+                    name -> "text".equals(name),
+                    maxAnalyzedOffset,
+                    limitToMaxAnalyzedOffset
+            );
+            highlighter.setFieldMatcher((name) -> "text".equals(name));
+            final Snippet[] snippets = highlighter.highlightField(getOnlyLeafReader(reader), topDocs.scoreDocs[0].doc, () -> rawValue);
+            assertEquals(expectedPassages.length, snippets.length);
+            for (int i = 0; i < snippets.length; i++) {
+                assertEquals(expectedPassages[i], snippets[i].getText());
+            }
+        } finally {
```

Comment:
You can use try-with-resources here I think?

Reply:
Thought about it but we need to tries since, something like that: https://gist.github.com/matriv/ca6ca5fe191c8e0b1e49af282ec90669

so I preferred the traditional way, but happy to change.